spec/factories/firm.rb

```diff
@@ -25,11 +25,19 @@
     longitude { Faker::Address.longitude.to_f.round(6) }
     status :independent
 
-    factory :onboarded_firm, traits: [:with_advisers, :with_offices], aliases: [:publishable_firm] do
-      advisers_count 1
+    transient do
       offices_count 1
     end
 
+    after(:create) do |firm, evaluator|
+      create_list(:office, evaluator.offices_count, firm: firm)
+      firm.reload
+    end
+
+    factory :onboarded_firm, traits: [:with_advisers] do
```

Comment:
I don't want this to grown more arms and legs, however :publishable_firm and :onboarded_firm are theoretically the same thing and if that makes sense we could bin one of them.